### DISCUSSION TITLE
Why does willUpdate exist?

### DISCUSSION THREAD

**Discussion by vdegenne** [lit/lit#3683] [General]
I was just wondering, why does `willUpdate` exist on the ReactiveElement? Isn't it just the same as:

```javascript
override update(_changedProperties) {
  // Do stuff before it really updates
  // ...
  super.update(_changedProperties);
}
```

?

**Comment by justinfagnani** [comment]
`willUpdate()` is very much like `updated()` - they both could be done with an override of `update()` and work before and after super. For client-side uses, they're mostly just conveniences. One difference with `willUpdate()` is that it runs before reactive controllers hostUpdate.

On the server, willUpdate is supposed to be a DOM-free callback to compute state so it can be called on the server.

**Reply by vdegenne** [reply]
Oh I see, `update` is not called on server? Thanks for your time
